packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -649,21 +649,37 @@ export class DataStores implements IDisposable {
      * This is called to update objects whose routes are unused. The unused objects are deleted.
      * @param unusedRoutes - The routes that are unused in all data stores in this Container.
      */
-    public updateUnusedRoutes(unusedRoutes: string[]) {
```

Comment:
`DataStores` needs to handle scenarios where it interacts with a deleted data store. Like when it gets an op or signal, etc. Not sure if you are planning to do that as a follow up but FYI.

Reply:
Also, we should have tests for this scenario.

Reply:
Yeah, I have it as a follow up, they're commented out in the tests right now

Reply:
AB#2383, if an op comes for a deleted datastore, it will be gone from the runtime, so the datastore context will need to be loaded first. Thus receiving ops/signals and loading should be the same thing.

Reply:
Ideally, yes. But its possible to get ops without loading the data store. Imagine, a client uses request pattern to get a sweep ready data store just before it becomes sweep ready and then sends an op for it. By the time the op is sequenced, the data store becomes sweep ready. Now, this op can be recevied by multiple clients and they should fail. This problem should go away when request pattern is gone, but for now, this is the only way to identify these issues.

Reply:
Ok, you have a point. I'll have to look into it more, that will be a separate PR.

Reply:
Okay, sounds good.